application-services/custom/camera-management/appcamera/evam.go

```diff
@@ -218,6 +221,43 @@ func (app *CameraManagementApp) getPipelineStatus(deviceName string) (interface{
 	return nil, nil
 }
 
+// processEdgeXEvent is the function that is called when an EdgeX event is received
+
+func (app *CameraManagementApp) processEdgeXEvent(_ interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
+	if data == nil {
+		app.lc.Errorf("processEdgeXEvent: was called without any data")
+		return false, nil
+	}
+
+	if app.config.AppCustom.DefaultPipelineName == "" || app.config.AppCustom.DefaultPipelineVersion == "" {
```

Comment:
We can check at startup, but then we would need to store a variable indicating the result, as we would need to check it here as well, so I am fine with this code the way it is, but I would change the error to be a warning or info.

Reply:
Also, I would move this check to after the other checks, right before you are about to start the pipeline. I know it feels better to put it up here, but due to the amount of device update calls we make, this would be printing a lot of messages.